### PR TITLE
[Guide] ES6: Compare var and let scopes. Fixes and enhancements.

### PR DESCRIPTION
**Pull request by AdrianSkar**
- Remove phantom image
- Remove notes for contributors
- Change explanation and hints so they don't provide the solution too early.
- Links to markdown + 2 more resources

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

### CODE REVIEW

**Review by bikingbadger — APPROVED** [review]
This looks good to me 👍. Thank you for your contribution!

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See requested changes.  Everything else looks good.

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (Manish-Giri) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]:
```suggestion
## Problem Explanation:
```
Please revert `Explanation` as @RandellDawson mentioned, and increase the heading level to h2 `##`.

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (Manish-Giri) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]:
```suggestion
## Code Explanation:
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (Manish-Giri) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]:
```suggestion
## Spoiler Alert!
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (Manish-Giri) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]:
```suggestion
*   _Remember that `let`'s scope is limited to the block, function or statement in which you declare it._
```

- Make the hint text italicized, following the same formatting as your first hint
- Add a "Hint 2" sub heading just above this hint (you can use the same formatting as Hint 1)

**Review by Manish-Giri — CHANGES_REQUESTED** [review]

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (AdrianSkar) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: I disagree. The original text gives away the solution so its not actually a hint.

Comment (Manish-Giri) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: > _Fix the code so that `i` declared in the if statement is a separate variable than `i` declared in the first line of the function._

@AdrianSkar This statement is not really clear. One could argue that since both variables are named `i`, they're the same variable, reused in a different scope. To avoid any confusion, please revert to the original sentence.

**Review by Manish-Giri — CHANGES_REQUESTED** [review]

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (AdrianSkar) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: Hi @Manish-Giri! I changed that text because I felt it "gives away" the solution but I didn't write the replacement, I used the [original exercise instructions](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords) (that's why I put it in italics). If you disagree we can just let it as it was, from my _newb_ perspective I like it better as in previous guide pages.

**Comment by thecodingaviator** [comment]
@Manish-Giri @RandellDawson How can we get this PRs ahead and merge it. Seems nothing is happening right now

**Comment by moT01** [comment]
@AdrianSkar @Manish-Giri @RandellDawson - This needs to get resolved since there's some larger PR's waiting on it - so lets figure out what we want. Perhaps adding a third part to the hints would help....
```
The log output should read...
'Block scope i is:  block scope'
'Function scope i is:  function scope'
```
In my opinion, it was fine how it originally was, or it's fine with these changes, I don't think there's a wrong way here... @AdrianSkar is right that, the original hints sort of gave away the solution. If I don't hear back in the near future from someone - I will probably just merge this so we can move on.

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: Continuing from @Manish-Giri , this is more confusing than ever. What do you mean by separate variable? Running on separate machines? If you really really want to do this, how about:
```suggestion
Change the code so that the variable `i` declared in the if block is separately scoped than the variable `i` declared at the beginning of the function
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (thecodingaviator) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: I agree with @AdrianSkar on this topic, the original text gives away the solution. What I propose is that we let the first hint be as has been suggested by the OP, but with the following formatting modifications:
```suggestion
Be certain not to use the `var` keyword anywhere in your code since variables declared with var are globally scoped.
```

Then we add a second hint with the text:
```text
Find `var` and replace with `let`.
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]:
```suggestion

```

Since we are in the process of phasing out links containing user controlled code, I'd appreciate if you remove this line

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (AdrianSkar) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: That's ok in my opinion. I put them in italics because it's a citation from the original instructions but that looks good too. Should I commit this changes?
Regarding the second hint it's actually the solution so not sure about that one. In any case I'm the inexperienced one so whatever you think it's best/could help someone with this lesson.

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (AdrianSkar) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: Those changes look perfect in my opinion. I didn't mean to confuse anyone, as I said I was trying to avoid giving users the solution in the instructions, I couldn't think of a better way to explain it and instead cited the original exercise instructions: 
image
If it's confusing maybe we could change those as well.

Comment (thecodingaviator) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: @RandellDawson @moT01 What do you think?

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (thecodingaviator) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: @AdrianSkar Before committing it, I'd say we get the approval of either of @RandellDawson / @moT01 / @Manish-Giri

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md
Comment (moT01) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: yes, sure @thecodingaviator @AdrianSkar, sounds good to me - someone make the commit please and we can give it a final review/merge

Comment (RandellDawson) [reply guide/english/certifications/javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords/index.md]: I tried to commit, but I get the message "You do not have permission to push to this repository. " for some reason.  @AdrianSkar I am fine with @thecodingaviator's suggestion.  Please commit it.  Thanks.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**
